### Table summarizing the codes available in this repository for the respective analysis/figures/tables
  
| Analysis/Figures/Tables                 | Section in the manuscript    | Name of the Github folder |	
| :------------------------: |:-------------:| :-------------: |
| Fig 1A |	Study cohort and proteogenomic profiling| Cohort_OmicsData_Overview |
| Data Pre-processing | STAR Method | Preprocessing |
| FFPE-Frozen data alignment| STAR Method | Preprocessing |
| Fig S1A-B | STAR Method | Preprocessing |
| Fig S1C | STAR Method | Preprocessing |
| Fig S1E-G | STAR Method | Preprocessing |
| Fig 1B	| Genomic alterations and their association with chemo-refractoriness| Response-Associatoin-Mut-LOH |
| Fig 1C	| Genomic alterations and their association with chemo-refractoriness| Response-Associatoin-Mut-LOH |
| Fig S1H	| Genomic alterations and their association with chemo-refractoriness| Response-Associatoin-Mut-LOH |
| Fig 1D	| Genomic alterations and their association with chemo-refractoriness| Response-Associatoin-Mut-LOH |
| Fig 1E | Transcripts, proteins, and CNVs associated with chemo-refractoriness | Response-Associatoin-CNV-RNA-protein |
| Fig 1F | Integrative analysis (CNV, RNA, protein) identifies chemotherapy response-associated genes | Response-Associatoin-CNV-RNA-protein |
| Fig 1G | Integrative analysis (CNV, RNA, protein) identifies chemotherapy response-associated genes | Response-Associatoin-CNV-RNA-protein |
| Fig S1I | Integrative analysis (CNV, RNA, protein) identifies chemotherapy response-associated genes | Response-Associatoin-CNV-RNA-protein |
| Fig 2A | Cis regulation between CNV, RNA, and protein differs between sensitive and refractory tumors | Cis-regulation |
| Fig 2B | Cis regulation between CNV, RNA, and protein differs between sensitive and refractory tumors | Cis-regulation |
| Fig S2A | Cis regulation between CNV, RNA, and protein differs between sensitive and refractory tumors | Cis-regulation |
| Fig S2B | Chromosome 17 loss-of-heterozygosity (Chr17-LOH) is associated with chemo-sensitivity | Chr17LOH |
| Fig 2C-D | Chromosome 17 loss-of-heterozygosity (Chr17-LOH) is associated with chemo-sensitivity | Chr17LOH |
| Fig 2E | Chromosome 17 loss-of-heterozygosity (Chr17-LOH) is associated with chemo-sensitivity | Chr17LOH |
| Fig 2F | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig S2C-D | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig 2G | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig 2H | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig S2E-G | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig 2I | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig S2H | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig S2I | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig 2J | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig S2J | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig 2K-N | Proteins associated with chemotherapy response in tumors stratified by Chr17-LOH, BRCA1/2, and TP53 mutations | Trans-regulation |
| Fig 3A | Genetic factors predictive of refractoriness | Prediction-refractoriness |
| Fig 3B | Genetic factors predictive of refractoriness | Prediction-refractoriness |
| Fig 3C | Genetic factors predictive of refractoriness | Prediction-refractoriness |
| Fig S3A-B | Genetic factors predictive of refractoriness | Prediction-refractoriness |
| Fig S3C | Genetic factors predictive of refractoriness | Prediction-refractoriness |
| Fig 3D | A protein panel predictive of refractoriness | Prediction-refractoriness |
| Fig S3D-E | A protein panel predictive of refractoriness | Prediction-refractoriness |
| Fig S3F-G | A proteogenomic panel predictive of refractoriness | Prediction-refractoriness |
| Fig 4A | Protein pathway enrichment analysis shows diverse processes associated with refractoriness | Response-Associatoin-Pathway |
| Fig S4A-B | Protein pathway enrichment analysis shows diverse processes associated with refractoriness | Response-Associatoin-Pathway |
| Fig 4B | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig 4C | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig S4C-D | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig 4D-F | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig S4E-F | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig 5A-D | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig S5A-E | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig 6A-E | Characterization of immune infiltration | Immune-infiltration |
| Fig S6A-D | Characterization of immune infiltration | Immune-infiltration |
| Fig 7A-F | Association of TGFβ, alt-EJ, and βAlt signatures with cluster and treatment response | TGFbeta |
| Fig S7A-E | Association of TGFβ, alt-EJ, and βAlt signatures with cluster and treatment response | TGFbeta |



























