### Table summarizing the codes available in this repository for the respective analysis/figures/tables
  
| Analysis/Figures/Tables                 | Section in the manuscript    | Name of the Github folder |	
| :------------------------: |:-------------:| :-------------: |
| Fig 1A |	Study cohort and proteogenomic profiling| Cohort_OmicsData_Overview |
| Data Pre-processing | STAR Method | Preprocessing |
| FFPE-Frozen data alignment| STAR Method | Preprocessing |
| Fig S1A,B,C,E,F,G | STAR Method | Preprocessing |
| Fig 1B,C,D	| Genomic alterations and their association with chemo-refractoriness| Response-Associatoin-Mut-LOH |
| Fig S1H	| Genomic alterations and their association with chemo-refractoriness| Response-Associatoin-Mut-LOH |
| Fig 1E | Transcripts, proteins, and CNVs associated with chemo-refractoriness | Response-Associatoin-CNV-RNA-protein |
| Fig 1F,G | Integrative analysis (CNV, RNA, protein) identifies chemotherapy response-associated genes | Response-Associatoin-CNV-RNA-protein |
| Fig S1I | Integrative analysis (CNV, RNA, protein) identifies chemotherapy response-associated genes | Response-Associatoin-CNV-RNA-protein |
| Fig 2A,B | Cis regulation between CNV, RNA, and protein differs between sensitive and refractory tumors | Cis-regulation |
| Fig S2A | Cis regulation between CNV, RNA, and protein differs between sensitive and refractory tumors | Cis-regulation |
| Fig S2B | Chromosome 17 loss-of-heterozygosity (Chr17-LOH) is associated with chemo-sensitivity | Chr17LOH |
| Fig 2C,D,E | Chromosome 17 loss-of-heterozygosity (Chr17-LOH) is associated with chemo-sensitivity | Chr17LOH |
| Fig 2F,G,H,I,J | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig S2C,D | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig S2E,F,G,H,I,J | Bi-allelic and functional TP53 inactivation is associated with chemo-sensitivity | TP53 |
| Fig 2K,L,M,N | Proteins associated with chemotherapy response in tumors stratified by Chr17-LOH, BRCA1/2, and TP53 mutations | Trans-regulation |
| Fig 3A,B,C | Genetic factors predictive of refractoriness | Prediction-refractoriness |
| Fig S3A,B,C | Genetic factors predictive of refractoriness | Prediction-refractoriness |
| Fig 3D | A protein panel predictive of refractoriness | Prediction-refractoriness |
| Fig S3D,E | A protein panel predictive of refractoriness | Prediction-refractoriness |
| Fig S3F,G | A proteogenomic panel predictive of refractoriness | Prediction-refractoriness |
| Fig 4A | Protein pathway enrichment analysis shows diverse processes associated with refractoriness | Response-Associatoin-Pathway |
| Fig S4A,B | Protein pathway enrichment analysis shows diverse processes associated with refractoriness | Response-Associatoin-Pathway |
| Fig 4B,C,D,E,F | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig S4C,D,E,F | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig 5A,B,C,D | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig S5A,B,C,D,E | Protein-based pathway scores reveal five distinct HGSOC tumor clusters that are validated in two independent patient cohorts and PDX models| Clustering|
| Fig 6A,B,C,D,E | Characterization of immune infiltration | Immune-infiltration |
| Fig S6A,B,C,D | Characterization of immune infiltration | Immune-infiltration |
| Fig 7A,B,C,D,E,F | Association of TGFβ, alt-EJ, and βAlt signatures with cluster and treatment response | TGFbeta |
| Fig S7A,B,C,D,E | Association of TGFβ, alt-EJ, and βAlt signatures with cluster and treatment response | TGFbeta |



























